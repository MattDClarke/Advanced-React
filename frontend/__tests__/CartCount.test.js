import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');
    // same as:
    // expect(container).toHaveTextContent('11');
    // Update the props
    rerender(<CartCount count={12} />);
    // wait for CSSTransitionGroup animation to complete
    // wait for ... ms
    // CSSTRansition group - incoming 12 component and outgoing 11 component briefly in DOM together
    expect(container.textContent).toBe('1211');
    await wait(400);
    // this method will timeout after 3s if it cant find the text
    // await screen.findByText('12');
    expect(container.textContent).toBe('12');
    expect(container).toMatchSnapshot();
    debug();
  });
});
