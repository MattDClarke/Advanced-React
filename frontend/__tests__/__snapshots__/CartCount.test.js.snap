// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 hXlggc"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 hXlggc"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
